pkg/server/server.go

```diff
@@ -53,6 +52,22 @@ const (
 	grpcDialTimeout = 5 * time.Second
 )
 
+func ServeUI(ctx context.Context, logger log.Logger, cfg UIConfig) {
+	isUIPortNotExits := cfg.Port == 0
+	if isUIPortNotExits {
+		logger.Error("ui server failed: no port specified")
```

Comment:
setting port should not be logged with error, could be a warning like `ui port unspecified, keeping it disabled`